src/static/js/chapter.js

```diff
@@ -0,0 +1,189 @@
+//Check if the screen meets minimum size requirements for Interactive figures
+//At the moment we base it on 600px break point matching CSS but it does not need to be the same
+function bigEnoughForInteractiveFigures() {
+  if (!(window.matchMedia('(max-width: 600px)').matches)) {
+    gtag('event', 'min-sheets-width', { 'event_category': 'user', 'event_label': 'true', 'value': 1 });
+    return true;
+  }
+  gtag('event', 'min-sheets-width', { 'event_category': 'user', 'event_label': 'enabfalseled', 'value': 0 });
+  console.log('Screen too small for interactive visuals');
+  return false;
+}
+
+//Data Save can be set to on, so let's check it
+function dataSaverEnabled() {
+  var dataSaver = false;
+  if ('connection' in navigator) {
+    dataSaver = navigator.connection.saveData;
+    if (dataSaver) {
+      console.log('DataSaver is enabled');
+      gtag('event', 'data-saver', { 'event_category': 'user', 'event_label': 'enabled', 'value': 1 });
+    } else {
+      gtag('event', 'data-saver', { 'event_category': 'user', 'event_label': 'not-enabled', 'value': 0 });
+    }
+  }
+
+  return dataSaver;
+}
+
+//iOS causes Google Sheets to create a 6000 by 3700 canvas, which annoyingly isn't supported by iOS!
+//Let's check if we have large Canvas support (annoyingly no API for this!)
+function highResolutionCanvasSupported() {
+
+  var largeCanvasSupported = false;
+
+  try {
+    // Set large sized canvas dimensions and draw test rectangle
+    var cvs = document ? document.createElement('canvas') : null;
+    var ctx = cvs && cvs.getContext ? cvs.getContext('2d') : null;
+    cvs.width = 6000;
+    cvs.height = 3700;
+    ctx.fillRect(5999, 3699, 1, 1);

```

Comment:
I think we may need to test this on low- or mid-range devices? It may be very processor intensive. I don't know, I'm just nervous.

Reply:
It's a fair point. Especially as Google Sheets appears to do some device detection and in Chrome only returns 1200 by 742 canvas sheets even on large screens, as opposed to iOS when it does this 6000 by 3700 sheet. Wish they handled this better for iOS as the canvas they return for it is not handled by it!

Anyway, I changed it to this:

```javascript
// Set large sized canvas dimensions and draw test rectangle
var cvs = document ? document.createElement('canvas') : null;
var ctx = cvs && cvs.getContext ? cvs.getContext('2d') : null;
var scale = window.devicePixelRatio || 1;
if (scale > 1) {
  cvs.width = 6000;
  cvs.height = 3700;
  ctx.fillRect(5999, 3699, 1, 1);
} else {
  cvs.width = 1200;
  cvs.height = 742;
  ctx.fillRect(1199, 741, 1, 1);
}
largeCanvasSupported = Boolean(ctx.getImageData(5999, 3699, 1, 1).data[3]) == 1;
```

And confirmed that still fixed it in iOS (which has a device pixel ratio of 2).

With that, and your `data-saver` re-order suggestion below I think that should address this concern. Plus I think overall we will be returning a lot less data for low end devices thanks to this change as we won't load sheets by default and will only do above test once per page at the start.

You agree?

Reply:
I do agree. As long as we test it. I have a Nokia 2 at work (for testing purposes), I can try and test it once this goes live?

Reply:
Good stuff. Any ideas what the low end devices are on Webpagetest? Will run a before and after on there too.

Reply:
Moto G1 has been suggested on the webperformance slack. Pages don't seem to render on that. Maybe that'll change after this PR :-)

Reply:
@mikegeyser if you could baseline production on your Nokia 2 that would be good to know so we have something to compare this too if we do go ahead with this. Try it on HTTP/2 chapter (only 1 Sheet chart), Accessibility (7 Sheets charts) and CSS (29 Sheets charts!!).